### PR TITLE
[System]: Renegotiation in AppleTls requires OS X 10.12+.  #9234.

### PR DESCRIPTION
**Pull request by baulig**
Fixes https://github.com/mono/mono/issues/9234

### CODE REVIEW

**Comment by baulig** [comment]
@monojenkins backport 2018-06

File: mcs/class/System/Mono.AppleTls/AppleTlsContext.cs [hunk 1]
```diff
@@ -341,10 +341,12 @@ void InitializeConnection ()
 				PeerDomainName = ServerName;
 			}
 
-			if (Options.AllowRenegotiation)
+			if (Options.AllowRenegotiation && Environment.OSVersion.Version >= MinRenegotiationVersion)
```
Comment (akoeplinger) [inline mcs/class/System/Mono.AppleTls/AppleTlsContext.cs]: will this work on iOS etc? I'm not sure which Environment.OSVersion we're getting there and the API is listed as iOS 10.0+, tvOS 10.0+, watchOS 3.0+.

shouldn't we just call CanRenegotiate? or is that just used on the server-side?

Comment (marek-safar) [reply mcs/class/System/Mono.AppleTls/AppleTlsContext.cs]: I think we need better "checking" method which does include all MONOTOUCH profiles. The option is supported https://developer.apple.com/documentation/security/sslsessionoption/ksslsessionoptionallowrenegotiation?language=objc

XI does it like https://github.com/xamarin/xamarin-macios/blob/54101252aca7e15a9e4fd04e98d1bfb49e411f05/src/IOSurface/IOSurfacePropertyKey.cs#L34-L43

**Comment by akoeplinger** [comment]
@monojenkins backport 2018-06

**Pull request merged**